modules/tinymce/src/core/main/ts/keyboard/PageUpDownKeys.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import Editor from '../api/Editor';
+import VK from '../api/util/VK';
+import * as InlineBoundariesNavigation from './InlineBoundariesNavigation';
+import * as MatchKeys from './MatchKeys';
+
+const executeKeydownOverride = (editor: Editor, evt: KeyboardEvent) => {
+  MatchKeys.execute([
+    { keyCode: VK.PAGE_UP, action: MatchKeys.action(InlineBoundariesNavigation.moveOutInlineBoundaries, editor, false) },
+    { keyCode: VK.PAGE_DOWN, action: MatchKeys.action(InlineBoundariesNavigation.moveOutInlineBoundaries, editor, true) }
+  ], evt).each((_) => {
+    evt.preventDefault();
+  });
+};
+
+const setup = (editor: Editor) => {
+  const blocker = (e) => e.stopImmediatePropagation();
```

Comment:
Why are we even listening to `keyup` here instead of `keydown` like all the other overrides, as that's going to cause various other issues such as flickering while it's moved initially and then repositioned? It's also going to cause a number of additional selection change events which isn't great for performance (though unlikely to be a bottleneck).

Additionally I'm very curious why we are blocking `NodeChange` as that's not obvious why that's needed? It also seems like something that's going to cause issues, as many things are hooked up to NodeChange and you're preventing it now, which in turn may block other logic such as Spell Checking running.

Given the above, this really feels like the wrong way to solve this and instead it should be listening and overriding `keydown` instead.

Reply:
I suggested it since we have no custom page up/down logic so we can't override the default as we do for the other things home/end that we do have custom code for. Also if the node change event is only blocked while the user is holding down the page up/down key it would have limited effects on other things. The alternative would be a flickering UI while caret move in and out of the inline boundary element. We could I guess skip page up/page down and implement that at a later date. Been thinking of how to do that for the content editable false re-implementation in rtc. But it's a lot of work since it need to be able to deal with layouts like tables etc.

Reply:
Fair enough. I guess that is okay for now, but If we do stick with this then we definitely need some comments to state why it's different and why we are blocking `NodeChange`. We ideally also should log a follow up JIRA to see if we can implement this in a better way by overriding keydown.

Reply:
Yes, I know it's a bit hacky but at this point it either that hack or not having the feature until later. Adding more context in comment here makes a lot of sense.

Reply:
updated